### PR TITLE
Added some comments and minor refactors.

### PR DESCRIPTION
**Pull request by ajaysubra**
# Description

Added some comments for readability.

# Check List

- [ ] Are you changing anything with the public API?
- [ ] Have you tested this change on real device?
- [ ] Are your changes backwards compatible with previous SDK Versions?
- [ ] Have you added unit test coverage for your changes?
- [ ] Have you verified that your changes are compatible with all the operating system version this SDK currently supports?

# Manual Test Plan

1.

# Supporting Materials

### CODE REVIEW

File: Sources/KlaviyoSwift/KlaviyoState.swift [hunk 1]
```diff
@@ -9,6 +9,9 @@ import AnyCodable
 import Foundation
 import UIKit
 
+typealias Metadata = KlaviyoAPI.KlaviyoRequest.KlaviyoEndpoint.PushTokenPayload.PushToken.Attributes.MetaData
```
[the remark this answers is not in the document]
Comment (evan-masseau) [reply Sources/KlaviyoSwift/KlaviyoState.swift:12]: super nit: something i found working on device metadata backend was that the word "metadata" is a bit overloaded, and so i tried to be more specific and always call it `DeviceMetadata` for a tiny bit more clarity

File: Sources/KlaviyoSwift/KlaviyoState.swift [hunk 2]
```diff
@@ -299,6 +321,9 @@ private func removeStateFile(at file: URL) {
     }
 }
 
+/// Loads SDK state from disk
+/// - Parameter apiKey: the API key that uniquily identiifies the company
```
Comment (evan-masseau) [inline Sources/KlaviyoSwift/KlaviyoState.swift:325]:
```suggestion
/// - Parameter apiKey: the API key that uniquely identiifies the company
```

File: Sources/KlaviyoSwift/StateManagement.swift [hunk 3]
```diff
@@ -26,26 +29,66 @@ enum RetryInfo: Equatable {
 }
 
 enum KlaviyoAction: Equatable {
+    /// Sets the API key to state. If the state is already initilized then the push token is moved over to the company with the API key provided in this action.
```
[the remark this answers is not in the document]
Comment (evan-masseau) [reply Sources/KlaviyoSwift/StateManagement.swift:32]:
```suggestion
    /// Sets the API key to state. If the state is already initialized then the push token is moved over to the company with the API key provided in this action.
```

File: Sources/KlaviyoSwift/StateManagement.swift [hunk 4]
```diff
@@ -26,26 +29,66 @@ enum RetryInfo: Equatable {
...
+    /// Loads the state from disk and carries over existing items from the queue. This emits `completeInitialization` at the end with the state loaded from disk.
     case initialize(String)
+
+    /// after the SDK is initllized creates an initial state from existing state from disk (if it exists) and queues up any tasks that are pending
```
Comment (evan-masseau) [inline Sources/KlaviyoSwift/StateManagement.swift:36]:
```suggestion
    /// after the SDK is initialized, creates an initial state from existing state from disk (if it exists) and queues up any tasks that are pending
```

File: Sources/KlaviyoSwift/StateManagement.swift [hunk 5]
```diff
@@ -26,26 +29,66 @@ enum RetryInfo: Equatable {
...
     case completeInitialization(KlaviyoState)
+
+    /// if initilized set the email else queue it up
```
Comment (evan-masseau) [inline Sources/KlaviyoSwift/StateManagement.swift:39]:
```suggestion
    /// if initialized, set the email else queue it up
```

File: Sources/KlaviyoSwift/StateManagement.swift [hunk 6]
```diff
@@ -26,26 +29,66 @@ enum RetryInfo: Equatable {
...
     case setEmail(String)
+
+    /// if initilized set the phone number else queue it up
```
Comment (evan-masseau) [inline Sources/KlaviyoSwift/StateManagement.swift:42]:
```suggestion
    /// if initialized set the phone number else queue it up
```

File: Sources/KlaviyoSwift/StateManagement.swift [hunk 7]
```diff
@@ -26,26 +29,66 @@ enum RetryInfo: Equatable {
...
     case setPhoneNumber(String)
+
+    /// if initilized set the external id else queue it up
```
Comment (evan-masseau) [inline Sources/KlaviyoSwift/StateManagement.swift:45]: lol i think you get the idea ^^

Comment (evan-masseau) [reply Sources/KlaviyoSwift/StateManagement.swift:45]:
```suggestion
    /// if initialized set the external id else queue it up
```

File: Sources/KlaviyoSwift/StateManagement.swift [hunk 8]
```diff
@@ -26,26 +29,66 @@ enum RetryInfo: Equatable {
...
+    /// called when the user wants to reset the existing profile from state
+    case resetProfile
+
+    /// dequeues requests that completed and contuinues to flush other requests if they exist.
     case dequeCompletedResults(KlaviyoAPI.KlaviyoRequest)
+
+    /// when the network connectivity change we want to use a different flush interval to flush out the pending requests
     case networkConnectivityChanged(Reachability.NetworkStatus)
+
+    /// flushes the queue say when the app is forgrounded or we come back to having network from not having
```
Comment (evan-masseau) [inline Sources/KlaviyoSwift/StateManagement.swift:60]:
```suggestion
    /// flushes the queue say when the app is foregrounded or we come back to having network from not having
```

**Review by evan-masseau — APPROVED** [review]
Looks good, i especially like the type aliases for readability! Just suggested typo fixes

**Review by ndurell — APPROVED** [review]
Approved. Great catch on the refactors!

File: Sources/KlaviyoSwift/StateManagement.swift (hunk 4 above)
Comment (ajaysubra) [reply Sources/KlaviyoSwift/StateManagement.swift:36]: 😢  I can't spell. Thanks!

Comment (evan-masseau) [reply Sources/KlaviyoSwift/StateManagement.swift]: haha no worries, i thought maybe your A key was broken

**Pull request merged**